src/indexWrapper.ts

```diff
@@ -35,21 +35,17 @@ server.on('connection', (ws: WebSocket, req: IncomingMessage) => {
     const address = socket.remoteAddress
     const port = socket.remotePort
     const addrPort = `${address}:${port}`
-    logger(`Connection made ${addrPort}`)
+    logger.info({ addrPort }, `New websocket connection`)
 
     const wsc = makeWsConnection(ws, addrPort)
     wsConnections[addrPort] = wsc
-  } catch (e) {
+  } catch (err) {
+    logger.error({ err, message: 'New websocket connection error ' })
     ws.close(1011, 'Internal error')
   }
 })
 
 const makeWsConnection = (ws: WebSocket, addrPort: string): WsConnection => {
-  const logger = (...args): void => {
-    const d = makeDate()
-    console.log(`${d} ${addrPort}`, ...args)
```

Comment:
New logger loses the context of which address:port was connected that caused the logging. I would remove all changes from this commit and just wrap the pino logging with this function in the WsConnection object.

Reply:
Ah, fixed this simply with `logger = pinoLogger.child({ addrPort })` and reverting back to injecting the logger into the `io` object.